implementation/src/main/java/io/smallrye/config/SecretKeys.java

```diff
@@ -4,14 +4,15 @@
 
 @SuppressWarnings("squid:S5164")
 public final class SecretKeys {
-    private static final ThreadLocal<Boolean> LOCKED = ThreadLocal.withInitial(() -> Boolean.TRUE);
+    private static final ThreadLocal<Boolean> LOCKED = new ThreadLocal<>();
 
     private SecretKeys() {
         throw new UnsupportedOperationException();
     }
 
     public static boolean isLocked() {
-        return LOCKED.get();
+        Boolean result = LOCKED.get();
+        return result == null ? true : result;
```

Comment:
You probably will want to fix doUnlocked too.

Reply:
From my POV the current code is fine, it's not leaking the value that is the problem, but leaking the value attached to a ThreadLocal that has a reference to the deployment ClassLoader. 

Is it intentional that both doLocked and doUnlocked do not check the current state, and will always reset to the opposite value?

Reply:
The current state is checked via `isLocked` afaict.

Reply:
ah, of course. I have pushed a fix